projects/storefrontlib/src/shared/services/trigger/services/render.strategy.ts

```diff
@@ -0,0 +1,54 @@
+import {
+  TriggerInlineMapping,
+  TriggerMapping,
+  TriggerOutletMapping,
+  TriggerUrlMapping,
+  TRIGGER_CALLER,
+} from '../config';
+
+export abstract class RenderStrategy {
+  // List of called references; only used for rendered elements
+  protected renderedCallers: TRIGGER_CALLER[] = [];
+
+  /**
+   * Render method to implement based on the strategy
+   *
+   * @param config Trigger configuration
+   */
+  abstract render(
```

Comment:
I was assuming that we need the actual trigger element as an input. For example to find the vcr for inline rendering, or to re-focus later on (out of scope for now) if the target is closed.